// Proto file containing commonly used messages

syntax = "proto3";

package ansys.api.edb.v1;

import public "google/protobuf/wrappers.proto";
import public "google/protobuf/empty.proto";

message ComplexMessage {
  double re = 1;
  double im = 2;
}

// Message representing a Value
message ValueMessage {
  string text = 1;
  oneof value {
    ComplexMessage constant = 2;
    EDBObjMessage expression = 3;
  }
}

// Message representing an arbitrary edb object
message EDBObjMessage {
  uint64 impl_ptr_address = 1;
  uint64 id = 2;
}

// Message containing a collection of edb objects 
message EDBObjCollectionMessage {
  repeated EDBObjMessage items = 1;
}

// Message representing an arbitrary edb object and a name
message EDBObjNameMessage {
  EDBObjMessage target = 1;
  string name = 2;
}
